src/obj_char_transfer.c

```diff
@@ -0,0 +1,872 @@
+        return NULL;
+    }
+
+    return &sObjCharTransferTasksManager->tasks[i];
+}
+
+static BOOL ObjCharTransferTask_ReserveAndTransfer(ObjCharTransferTask *task) {
+    BOOL ret = TRUE;
+    if (task->isTransferred) {
+        task->state = OBJ_CHAR_TRANSFER_TASK_2;
+    } else {
+        task->state = OBJ_CHAR_TRANSFER_TASK_1;
+        if (!task->transferFromTail) {
+            ret = ObjCharTransferTask_ReserveFromHead(task);
+        } else {
+            ret = ObjCharTransferTask_ReserveFromTail(task);
+        }
+    }
+    return ret;
+}
+
+static BOOL ObjCharTransferTask_ReserveFromHead(ObjCharTransferTask *task) {
+    u32 offsetMain;
+    u32 offsetSub;
+    BOOL ret = ObjCharTransfer_TryGetDestVramOffsets(task->charData->szByte, task->vram, &offsetMain, &offsetSub);
+    if (ret) {
+        ObjCharTransferTask_SetBaseAddrs(task, offsetMain, offsetSub);
+        ObjCharTransfer_ReserveVramSpace(task->charData->szByte, task->vram);
+    }
+    return ret;
+}
+
+static BOOL ObjCharTransferTask_ReserveFromTail(ObjCharTransferTask *task) {
+    u32 offsetMain;
+    u32 offsetSub;
+    u32 sizeMain;
+    u32 sizeSub;
+    if (!getBlockNumAndFreeSpaceForTransfer(task->vram, &offsetMain, &offsetSub, task->charData->szByte, &sizeMain, &sizeSub)) {
+        return FALSE;
+    }
+    ObjCharTransferTask_UpdateBaseAddrs(task, offsetMain, offsetSub);
+    task->needResetBlockTransferFlags = TRUE;
+    task->sizeMain = sizeMain;
+    task->sizeSub = sizeSub;
+    ObjCharTransferTask_G2dLoadImageMapping(task);
+    reserveTransferBlocksByVramOffsetAndSize(task->vram, offsetMain, offsetSub, sizeMain, sizeSub);
+    return TRUE;
+}
+
+static void ObjCharTransferTask_Reset(ObjCharTransferTask *task) {
+    if (task->needResetBlockTransferFlags) {
+        ObjCharTransferTask_ResetBlockTransferFlags(task);
+    }
+    ObjCharTransferTask_Init(task);
+}
+
+static ObjCharTransferTask *ObjCharTransfer_GetTaskByID(int resId) {
+    for (int i = 0; i < sObjCharTransferTasksManager->max; ++i) {
+        if (sObjCharTransferTasksManager->tasks[i].resId == resId) {
+            return &sObjCharTransferTasksManager->tasks[i];
+        }
+    }
+
+    return NULL;
+}
+
+static GXOBJVRamModeChar ObjCharTransferTask_SetMappingTypeFromHW(ObjCharTransferTask *task, NNS_G2D_VRAM_TYPE vram) {
+    if (vram == NNS_G2D_VRAM_TYPE_2DMAIN) {
+        task->charData->mapingType = GX_GetOBJVRamModeChar();
+    } else {
+        task->charData->mapingType = GXS_GetOBJVRamModeChar();
+    }
+    return task->charData->mapingType;
+}
+
+static void ObjCharTransferTask_SetBaseAddrs(ObjCharTransferTask *task, u32 offsetMain, u32 offsetSub) {
+    task->baseAddrMain = offsetMain;
+    task->baseAddrSub = offsetSub;
+    if (!task->isTransferred) {
+        ObjCharTransferTask_G2dLoadImageMapping(task);
+    }
+}
+
+static BOOL getBlockNumAndFreeSpaceForTransfer(int vram, u32 *blockNumMain, u32 *blockNumSub, u32 size, u32 *freeSpaceMain, u32 *freeSpaceSub) {
```

Comment:
I didn't see a static declaration at the top of the file for this, though I could just be blind
also needs a prefix for this func

Reply:
you are blind. also, i omitted the prefix and used camelCase because this function is local and operates only on primitive types

Reply:
ehhhhhh, it's still part of object transfer tbh....